src/loki/worker.cc

```diff
@@ -92,6 +93,8 @@ void loki_worker_t::parse_costing(Api& api, bool allow_none) {
     throw valhalla_exception_t{124};
   }
 
+  check_hierarchy_limits(api);
```

Comment:
We parse costing options in loki, so I'd say we do the same here, even though hierarchy limits are only used in thor. 

The high level idea is: 
  - parse it off the json inside `ParseBaseCostingOptions` 
  - validate it inside loki worker's `parse_costing`, since the worker knows about the service limits (are users allowed to modify it, what are the max allowed values)
  - set it in `DynamicCost` only if there is anything on the protobuf options and expose a flag that tells algorithms whether there are customized hierarchy limits
  - in the  algorithms that use hierarchy limits, prefer user provided hierarchy limits, fall back to config

Reply:
talked about this in a video call but basically to  really support all the different algorithms having their own service limits we'll need thor worker to parse these service limits and do the check there where it decides which algorithm will be used